packages/sanity/src/form/inputs/Slug/SlugInput.tsx

```diff
@@ -15,26 +23,36 @@ export type Slug = {
 
 export type SlugInputProps = ObjectInputProps<Slug, SlugSchemaType>
 
-function getNewFromSource(
-  source: string | Path | SlugSourceFn,
+function getSlugSourceContext(
   valuePath: Path,
-  document: SanityDocument
-) {
+  document: SanityDocument,
+  context: SlugContext
+): SlugSourceContext {
   const parentPath = valuePath.slice(0, -1)
   const parent = PathUtils.get(document, parentPath) as SlugParent
+  return {parentPath, parent, ...context}
+}
+
+function getNewFromSource(
```

Comment:
suggestion: make this an async function? That way we make sure that any (current or future) sync errors triggered by either this function or the slugifier call are thrown on the returned promise. As a bonus, the Promise.resolve call in here can be removed too.

Reply:
Will do. Funny seeing eslint fight itself:

![image](https://user-images.githubusercontent.com/835514/193681052-28a84014-c70d-4c52-8ed9-fa8a1960fadc.png)

![image](https://user-images.githubusercontent.com/835514/193681113-abea49e9-5137-4616-a7de-78a05ce6159d.png)

🤷

Reply:
Seems like we should turn off the `require-await` rule or replace it with `@typescript/require-await` which uses type information: https://typescript-eslint.io/rules/require-await/